mesh/nurbs.cpp

```diff
@@ -2369,13 +2371,22 @@ void NURBSExtension::Load(std::istream &input, bool spacing)
          patches[p] = new NURBSPatch(input);
       }
 
-      NumOfKnotVectors = 0;
-      for (int i = 0; i < patchTopo->GetNEdges(); i++)
-         if (NumOfKnotVectors < KnotInd(i))
-         {
-            NumOfKnotVectors = KnotInd(i);
-         }
-      NumOfKnotVectors++;
+      if (Dimension() == 1)
+      {
+         // In 1D, each patch has its own (possibly different) KnotVector.
+         // The number of unique KnotVectors is equal to the number of patches.
+         NumOfKnotVectors = patches.Size();
+      }
+      else
+      {
+         NumOfKnotVectors = 0;
+         for (int i = 0; i < patchTopo->GetNEdges(); i++)
+            if (NumOfKnotVectors < KnotInd(i))
+            {
+               NumOfKnotVectors = KnotInd(i);
+            }
+         NumOfKnotVectors++;
+      }
```

Comment:
In PR #5123 @dylan-copeland suggested that (inadvertently?) allow the use case where the same kv is reused. This addition seems to not allow that for 1D.

Could there be use case where you would like to have different line segments have the same knot?

Does the code work without this addition? If it does I would suggest to remove it.
If not? Why not?